OSIPI/pypi
# Poetry

This PR introduces Poetry as the dependency manager for our project and refactors the GitHub Actions workflow to use Poetry for linting and testing.

Changes include:
- Adding `poetry`  to `pyproject.toml` and `poetry.lock` for Poetry dependency management.
- Updating the GitHub Actions workflow to install dependencies with Poetry.
- Refactoring the linting and pytest steps to use Poetry.

## How to Test
1. Ensure that Poetry is installed locally.
2. Run `poetry install` to install dependencies.
3. Activate the Virtual Environment `poetry shell`
4. To run the application `poetry run file.py`
5. To run the tests, use: `poetry run pytest`
